frontend/src/modules/projectinfo/components/ProjectInfo.js

```diff
@@ -0,0 +1,55 @@
+/* @flow */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import './ProjectInfo.css';
+
+import * as navigation from 'core/navigation';
+import * as project from 'core/project';
+
+import InfoPanel from './InfoPanel';
+
+import type { NavigationParams } from 'core/navigation';
+import type { ProjectState } from 'core/project';
+
+
+type Props = {|
+    parameters: NavigationParams,
+    project: ProjectState,
+|};
+
+type InternalProps = {|
+    ...Props,
+    dispatch: Function,
+|};
+
+
+/**
+ * Render the current project's completion status and its info panel.
+ */
+export class ProjectInfoBase extends React.Component<InternalProps> {
+    render() {
+        if (
+            this.props.project.fetching ||
+            !this.props.project.name ||
```

Comment:
You mean !this.props.project.**info** ||? There's no point in showing the icon and empty panel if there's no project info.

Reply:
I did mean `name` here, as my intent was just to check if the data hadn't been fetched yet, but you're right! It also makes the component even more self-contained. :)